client-sdk/go/crypto/signature/context.go

```diff
@@ -1,27 +1,68 @@
 package signature
 
 import (
+	"encoding/hex"
+
 	"github.com/oasisprotocol/oasis-core/go/common"
 	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
 )
 
 const chainContextSeparator = " for chain "
 
-// Context is the chain domain separation context.
-type Context string
+// SignContext stores runtime ID, consensus chain context and tx-specific details
+// as defined in ADR 14 appropriate for CBOR encoding.
```

Comment:
Maybe consider separating the context in-memory representation from the serializable struct. E.g. not conflate the two, but define a separate ADR 14-compatible struct and provide a conversion function. This enables you to have more rich types in `Context` if needed.